### PR TITLE
docs(NgClass): update docs, add example

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**

### CODE REVIEW

File: modules/angular2/src/core/directives/ng_class.ts [hunk 1]
```diff
@@ -16,23 +16,40 @@ import {
...
+ * - `Object` - each key corresponds to a CSS class name while values are interpreted as expressions
+ * evaluating to `Boolean`. If a given expression evaluates to `true` a corresponding CSS class
+ * is added - otherwise it is removed.
  *
- * # Example:
+ * While the `NgClass` directive can interpret expressions evaluating to `string`, `Array`
+ * or `Object`, the `Object`-based version is the most often used and has an advantage of keeping
+ * all the CSS class names in a template.
+ *
+ * ### Example
```
Comment (IgorMinar) [inline modules/angular2/src/core/directives/ng_class.ts]: please add plunker link

File: modules/angular2/src/core/directives/ng_class.ts [hunk 2]
```diff
@@ -16,23 +16,40 @@ import {
 } from 'angular2/src/core/facade/collection';
 
 /**
- * Adds and removes CSS classes based on an {expression} value.
+ * The `NgClass` directive can be used to conditionally add and remove CSS classes on
```
Comment (IgorMinar) [inline modules/angular2/src/core/directives/ng_class.ts]: please don't use passive voice. rewrite as "Use `NgClass` directive to conditionally add and remove CSS .."

File: modules/angular2/src/core/directives/ng_class.ts [hunk 3]
```diff
@@ -16,23 +16,40 @@ import {
...
- *     Please check errors.
- * </div>
+ * import {Component, View, Input, NgClass} from 'angular2/angular2';
+ *
+ * @Component({
+ *     selector: 'toggle-button'
+ * })
+ * @View({
+ *     template: `
+ *     <div [ng-class]="{active: isOn, disabled: isDisabled}" (click)="isOn = !isOn">
```
Comment (IgorMinar) [inline modules/angular2/src/core/directives/ng_class.ts]: can you add an inline style sheet with active and disabled classes?

**Comment by IgorMinar** [comment]
@pkozlowski-opensource can you please clean this up and merge?

**Comment by pkozlowski-opensource** [comment]
@IgorMinar was trying to do so today but was hitting pbs with inlined styles in alpha.39. Need to investigate more to confirm that this is a bug. But yeh, I didn't forget this one, just wasn't able to make the plunk work with inline styles. Will continue tomorrow.

**Comment by pkozlowski-opensource** [comment]
Moving to alpha-41 as I won't have time to finish this one today

**Pull request closed without merging**